assets/src/infrastructure/ui/display/confirmDelete/ConfirmDelete.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import {
+	AlertDialog,
+	AlertDialogBody,
+	AlertDialogFooter,
+	AlertDialogHeader,
+	AlertDialogContent,
+	AlertDialogOverlay,
+	Button,
+} from '@chakra-ui/core';
+import { EspressoButton, EspressoButtonProps } from '@application/ui/input';
+import { __ } from '@wordpress/i18n';
+
+interface ConfirmDeleteProps {
+	buttonProps: EspressoButtonProps;
+	onConfirm: VoidFunction;
+	title?: string;
+}
+
+const ConfirmDelete: React.FC<ConfirmDeleteProps> = ({ buttonProps, onConfirm, ...props }) => {
+	const [isOpen, setIsOpen] = React.useState(false);
+	const cancelRef = React.useRef();
+	const onClose = () => setIsOpen(false);
+	const onDelete = () => {
+		onClose();
+		onConfirm();
+	};
+	const title = props.title || __('Are you sure you want to delete this?');
+
+	return (
+		<>
+			<EspressoButton {...buttonProps} onClick={() => setIsOpen(true)} />
+
+			<AlertDialog isOpen={isOpen} leastDestructiveRef={cancelRef} onClose={onClose}>
+				<AlertDialogOverlay />
+				<AlertDialogContent>
+					{buttonProps?.tooltip && (
+						<AlertDialogHeader fontSize='lg' fontWeight='bold' textTransform='capitalize'>
+							{buttonProps.tooltip}
+						</AlertDialogHeader>
+					)}
+					<AlertDialogBody>{title}</AlertDialogBody>
+					<AlertDialogFooter>
+						<Button ref={cancelRef} onClick={onClose}>
+							{__('No')}
+						</Button>
+						<Button variantColor='red' onClick={onDelete} ml={3}>
+							{__('Yes')}
+						</Button>
+					</AlertDialogFooter>
+				</AlertDialogContent>
+			</AlertDialog>
+		</>
+	);
+};
+
+export default ConfirmDelete;
```

Comment:
IMHO, `ConfirmDelete` does not belong to infrastructure. It belongs to `application` but will use components from infrastructure.

Reply:
that will create another level of abstraction, thus making reasoning a bit more convoluted

Reply:
a good example of abstraction is BaseInput and then other inputs are composed based on that base components

Reply:
help me understand the composition between `infrastructure` and `application` layers

Reply:
done, also added the variation I was talking about yesterday: 
https://github.com/eventespresso/event-espresso-core/pull/2682/commits/532bca0daeb3191b8ee366122cca0532c7eee742
*** 
next will add `ConfirmClose` variation